test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -904,8 +919,8 @@ public XContentBuilder shuffleXContent(XContentBuilder builder, String... except
      * recursive shuffling behavior can be made by passing in the names of fields which
      * internally should stay untouched.
      */
-    public XContentBuilder shuffleXContent(XContentParser parser, boolean prettyPrint, String... exceptFieldNames) throws IOException {
-        // use ordered maps for reproducibility
+    protected static XContentBuilder shuffleXContent(XContentParser parser, boolean prettyPrint, String... exceptFieldNames) throws IOException {
+        //TODO why do we need sorted map if we later sort the keys right before shuffling them? That should be enough?
```

Comment:
I think this is right, we don't need it. It might be that we introduced the key sorting later, I remember that the main concern was reproducibility with the test seed, but that should be taken care of by sorting the keys before shuffling.